# JavaBeans Coffee Sales Analysis

Sales Data Analysis for JavaBeans coffee chain between 2021-2022.

**Steps:**
1. Create a DataFrame with sales data
2. Calculate total sales and year-over-year growth (YoY)
3. Identify top/bottom performers and provide business insights

In [1]:
import pandas as pd

In [2]:
# Create a DataFrame with beverage sales data between 2021 and 2022
coffee_sales = pd.DataFrame({
    '2021 Sales': [562, 812, 426, 852],
    '2022 Sales': [623, 925, 384, 756]
}, index=['Americano', 'Latte', 'Espresso', 'Cappuccino'])

# Display the DataFrame
print("JavaBeans Coffee Sales Data:")
print(coffee_sales)
print(f"\nDataFrame shape: {coffee_sales.shape[0]} beverages, {coffee_sales.shape[1]} years")

JavaBeans Coffee Sales Data:
            2021 Sales  2022 Sales
Americano          562         623
Latte              812         925
Espresso           426         384
Cappuccino         852         756

DataFrame shape: 4 beverages, 2 years


In [3]:
# Calculate total sales across both years for each beverage
coffee_sales['Total Sales'] = coffee_sales['2021 Sales'] + coffee_sales['2022 Sales']

print("Total Sales (2021-2022):")
print(coffee_sales['Total Sales'].sort_values(ascending=False))

Total Sales (2021-2022):
Latte         1737
Cappuccino    1608
Americano     1185
Espresso       810
Name: Total Sales, dtype: int64


In [4]:
# Calculate YoY growth rate: (2022 Sales - 2021 Sales) / 2021 Sales * 100
coffee_sales['YoY Growth %'] = ((coffee_sales['2022 Sales'] - coffee_sales['2021 Sales']) / coffee_sales['2021 Sales'] * 100).round(2)

print("\nYear-over-Year Growth Rate:")
print(coffee_sales[['2021 Sales', '2022 Sales', 'YoY Growth %']].sort_values('YoY Growth %', ascending=False))


Year-over-Year Growth Rate:
            2021 Sales  2022 Sales  YoY Growth %
Latte              812         925         13.92
Americano          562         623         10.85
Espresso           426         384         -9.86
Cappuccino         852         756        -11.27


In [5]:
# Best and Worst Sellers for 2021
best_2021 = coffee_sales['2021 Sales'].idxmax()
best_2021_units = coffee_sales['2021 Sales'].max()
worst_2021 = coffee_sales['2021 Sales'].idxmin()
worst_2021_units = coffee_sales['2021 Sales'].min()

print("2021 Performance:")
print(f"  Best-seller: {best_2021} ({best_2021_units} units)")
print(f"  Least-seller: {worst_2021} ({worst_2021_units} units)")
print(f"  Difference: {best_2021_units - worst_2021_units} units")

# Best and Worst Sellers for 2022
best_2022 = coffee_sales['2022 Sales'].idxmax()
best_2022_units = coffee_sales['2022 Sales'].max()
worst_2022 = coffee_sales['2022 Sales'].idxmin()
worst_2022_units = coffee_sales['2022 Sales'].min()

print("\n2022 Performance:")
print(f"  Best-seller: {best_2022} ({best_2022_units} units)")
print(f"  Least-seller: {worst_2022} ({worst_2022_units} units)")
print(f"  Difference: {best_2022_units - worst_2022_units} units")

2021 Performance:
  Best-seller: Cappuccino (852 units)
  Least-seller: Espresso (426 units)
  Difference: 426 units

2022 Performance:
  Best-seller: Latte (925 units)
  Least-seller: Espresso (384 units)
  Difference: 541 units


In [6]:
# Display complete analysis table
print("\nComplete Analysis:")
print(coffee_sales.round(2))

# Calculate overall statistics
total_2021 = coffee_sales['2021 Sales'].sum()
total_2022 = coffee_sales['2022 Sales'].sum()
overall_growth = ((total_2022 - total_2021) / total_2021 * 100)

print(f"\nOverall Business Metrics:")
print(f"  2021 Total Sales: {total_2021} units")
print(f"  2022 Total Sales: {total_2022} units")
print(f"  Overall Growth: {overall_growth:.2f}%")


Complete Analysis:
            2021 Sales  2022 Sales  Total Sales  YoY Growth %
Americano          562         623         1185         10.85
Latte              812         925         1737         13.92
Espresso           426         384          810         -9.86
Cappuccino         852         756         1608        -11.27

Overall Business Metrics:
  2021 Total Sales: 2652 units
  2022 Total Sales: 2688 units
  Overall Growth: 1.36%


In [7]:
print("STRATEGIC ANALYSIS FOR JAVABEANS")

print("\n1. LATTE - Growing Fast")
print(f"   Up {coffee_sales.loc['Latte', 'YoY Growth %']:.2f}% from last year")
print(f"   Best seller now with {coffee_sales.loc['Latte', '2022 Sales']:.0f} units in 2022")
print("   People like this product and the demand is doing really well")
print("   Should try making different flavors (seasonal stuff)")

print("\n2. CAPPUCCINO - Losing Ground")
print(f"   Was #1 in 2021 ({coffee_sales.loc['Cappuccino', '2021 Sales']:.0f} units)")
print(f"   Now down to {coffee_sales.loc['Cappuccino', '2022 Sales']:.0f} units in 2022")
print(f"   Lost {coffee_sales.loc['Cappuccino', '2022 Sales'] - coffee_sales.loc['Cappuccino', '2021 Sales']:.0f} units (-{abs(coffee_sales.loc['Cappuccino', 'YoY Growth %']):.1f}%)")
print("   Need to figure out why customers switched to Lattes")
print("   Run promotions to get customers back")

print("\n3. AMERICANO - Steady Performer")
print(f"   Growing {coffee_sales.loc['Americano', 'YoY Growth %']:.2f}% from last year")
print(f"   Has {coffee_sales.loc['Americano', '2022 Sales']:.0f} units sold in 2022")
print("   Reliable product")

print("\n4. ESPRESSO - Not Working")
print(f"   Down {abs(coffee_sales.loc['Espresso', 'YoY Growth %']):.1f}% from last year")
print(f"   Only {coffee_sales.loc['Espresso', '2022 Sales']:.0f} units (worst performer)")
print("   Could use a rebrand")

print("WHAT THIS MEANS:")

print("\n1. Market is shifting toward milk-based drinks")
print("   Latte is taking off, Cappuccino is falling")
print("   Keep focusing on stuff like Latte")

print("\n2. Only Latte is doing well - risky situation")
print("   If Latte stops selling, business has a problem")
print("   Need to fix Cappuccino or make new products")

print("\n3. Overall growth is only 1.36% - that's pretty slow")
print("   Need better ways to grow the business")
print("   Fix the products that are failing")

print("ACTION ITEMS:")
print("\n1. Create new Latte flavors (seasonal, limited edition)")
print("2. Ask customers why they're switching from Cappuccino")
print("3. Run promotions on Cappuccino and Espresso")
print("4. Decide if Espresso is worth keeping")
print("5. Maybe raise Latte prices (it's popular, people will pay more)")

STRATEGIC ANALYSIS FOR JAVABEANS

1. LATTE - Growing Fast
   Up 13.92% from last year
   Best seller now with 925 units in 2022
   People like this product and the demand is doing really well
   Should try making different flavors (seasonal stuff)

2. CAPPUCCINO - Losing Ground
   Was #1 in 2021 (852 units)
   Now down to 756 units in 2022
   Lost -96 units (-11.3%)
   Need to figure out why customers switched to Lattes
   Run promotions to get customers back

3. AMERICANO - Steady Performer
   Growing 10.85% from last year
   Has 623 units sold in 2022
   Reliable product

4. ESPRESSO - Not Working
   Down 9.9% from last year
   Only 384 units (worst performer)
   Could use a rebrand
WHAT THIS MEANS:

1. Market is shifting toward milk-based drinks
   Latte is taking off, Cappuccino is falling
   Keep focusing on stuff like Latte

2. Only Latte is doing well - risky situation
   If Latte stops selling, business has a problem
   Need to fix Cappuccino or make new products

3. Overall gr

## Conclusion

JavaBeans is growing 1.36% overall, but the data shows some clear trends:

- **Latte is the winner** - up 13.92%, now the top seller
- **Cappuccino fell from 1st place** - down 11.27%, lost a lot of customers
- **Espresso is struggling** - down 9.86%, the worst performer
- **Americano is steady** - up 10.85%, reliable product

The main takeaway: customers want milk-based drinks. JavaBeans should focus on Lattes, figure out why Cappuccinos are losing customers, and decide what to do with Espresso. The business is depending on the success of Lattes.